catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/SceneTransitionWithSoundBrickStageTest.java

```diff
@@ -0,0 +1,169 @@
+import static junit.framework.TestCase.assertFalse;
+
+import static org.catrobat.catroid.common.Constants.SOUND_DIRECTORY_NAME;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+@RunWith(AndroidJUnit4.class)
+public class SceneTransitionWithSoundBrickStageTest {
+
+	private ScriptEvaluationGateBrick lastBrickFirstScript;
+	private ScriptEvaluationGateBrick lastBrickSecondScript;
+
+	private String firstSceneName;
+	private Script secondScript;
+
+	private File soundFile;
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION,
+			SpriteActivity.FRAGMENT_SOUNDS);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() throws IOException {
+		TestUtils.deleteProjects(getClass().getSimpleName());
+		if (soundFile != null && soundFile.exists()) {
+			soundFile.delete();
+		}
+	}
+
+	@Test
+	public void testStopSoundOnSceneTransition() {
+		MediaPlayer mediaPlayer = startAndGetMediaplayer(lastBrickSecondScript);
+		assertFalse(mediaPlayer.isPlaying());
+	}
+
+	@Test
+	public void testContinueSoundAfterSceneTransition() {
+		secondScript.addBrick(new SceneTransitionBrick(firstSceneName));
+		MediaPlayer mediaPlayer = startAndGetMediaplayer(lastBrickFirstScript);
+		assertTrue(mediaPlayer.isPlaying());
+	}
+
+	@Test
+	public void testContinueSoundDoesNotStartFromBeginning() {
+		secondScript.addBrick(new SceneTransitionBrick(firstSceneName));
+		MediaPlayer mediaPlayer = startAndGetMediaplayer(lastBrickFirstScript);
+		assertTrue(mediaPlayer.isPlaying());
+		assertTrue(mediaPlayer.getCurrentPosition() > 0);
+	}
+
+	private MediaPlayer startAndGetMediaplayer(ScriptEvaluationGateBrick scriptBrick) {
+		onView(withId(R.id.button_play))
+				.perform(click());
+		scriptBrick.waitUntilEvaluated(3000);
+		List<MediaPlayer> mediaPlayers = SoundManager.getInstance().getMediaPlayers();
+		assertEquals(1, mediaPlayers.size());
+		return mediaPlayers.get(0);
+	}
+
+	private void createProject() throws IOException {
+		Project project = new Project(ApplicationProvider.getApplicationContext(), getClass().getSimpleName());
+		Script script = new StartScript();
+		Sprite sprite = project.getDefaultScene().getBackgroundSprite();
+		XstreamSerializer.getInstance().saveProject(project);
+
+		firstSceneName = project.getDefaultScene().getName();
+		Scene secondScene = new Scene("Scene 2", project);
+
+		sprite.addScript(script);
+		project.getDefaultScene().addSprite(sprite);
+		ProjectManager.getInstance().setCurrentProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		project.addScene(secondScene);
+
+		script.addBrick(new PlaySoundBrick());
+		script.addBrick(new WaitBrick(500));
```

Comment:
it is necessary to let a sound play for a short time so that it is possible to check that it did not start from the beginning when continuing a scene.